packages/vscode-docker-registries/src/clients/GenericRegistryV2/GenericRegistryV2DataProvider.ts

```diff
@@ -0,0 +1,153 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { RegistryV2DataProvider, V2Registry, V2RegistryRoot, V2Repository, V2Tag } from '../RegistryV2/RegistryV2DataProvider';
+import { BasicOAuthProvider } from '../../auth/BasicOAuthProvider';
+import { GenericRegistryV2WizardContext, GenericRegistryV2WizardPromptStep } from './GenericRegistryV2WizardPromptStep';
+import { RegistryWizard } from '../../wizard/RegistryWizard';
+import { RegistryWizardSecretPromptStep, RegistryWizardUsernamePromptStep } from '../../wizard/RegistryWizardPromptStep';
+import { isContextValueRegistryItem } from '../../contracts/RegistryItem';
+
+const GenericV2StorageKey = 'GenericV2ContainerRegistry';
+const TrackedRegistriesKey = `${GenericV2StorageKey}.TrackedRegistries`;
+
+interface GenericV2RegistryRoot extends V2RegistryRoot {
+    readonly additionalContextValues: ['genericRegistryV2Root'];
+}
+
+export interface GenericV2Registry extends V2Registry {
+    readonly additionalContextValues: ['genericRegistryV2Registry'];
+}
+
+interface GenericV2RegistryTag extends V2Tag {
+    readonly additionalContextValues: ['genericRegistryV2Tag'];
+}
+
+export function isGenericV2Registry(item: unknown): item is GenericV2Registry {
+    return isContextValueRegistryItem(item) && item.additionalContextValues?.includes('genericRegistryV2Registry') === true;
+}

```

Comment:
We can maybe replace all this with overriding `getChildren` and attaching a `generic` context value

Reply:
that sounds good to me